Mathlib/Data/List/Chain.lean

```diff
@@ -194,6 +194,26 @@ theorem chain'_append_cons_cons {b c : α} {l₁ l₂ : List α} :
     Chain' R (l₁ ++ b :: c :: l₂) ↔ Chain' R (l₁ ++ [b]) ∧ R b c ∧ Chain' R (c :: l₂) := by
   rw [chain'_split, chain'_cons]
 
+theorem chain'_iff_all_append_cons_cons {l : List α} : Chain' R l ↔
```

Comment:
I think "forall" is used more often than "all" to denote `∀` in theorem names. 

One possible name would be `chain'_iff_forall_rel_of_append_cons_cons` - maybe @YaelDillies has suggestions for a better name?

Reply:
I think this is a hard one to name, but your name is better than the current one, yes